### DISCUSSION TITLE
genericLogTopicFrequencies

### DISCUSSION THREAD

**Discussion by Williamwenda** [USC-ACTLab/crazyswarm#355] [Q&A] [answered]
Hello, 

I am new to the crazyswarm project. After the setup, everything seems to be working fine. However, I have a few questions about the frequencies for generic log topics. 

I have one CF and one crazyradio connected to my PC and I wish to log 5 float variables. The launch file I am using is the "hover_swarm.launch"

If I put them into one log topic with 
      genericLogTopics: ["log1"]
      genericLogTopicFrequencies: [10]
      genericLogTopic_log1_Variables: ["tdoa3.d1-2","tdoa3.snr_1", "tdoa3.powerdiff_1", "tdoa3.snr_2", "tdoa3.powerdiff_2"]
, the frequency of  "/cf1/log1"  in ROS topic is around 70 Hz.

If I separate them into two log topics with
      genericLogTopics: ["log1", "log2"]
      genericLogTopicFrequencies: [10, 10]
      genericLogTopic_log1_Variables: ["tdoa3.d1-2"]
      genericLogTopic_log2_Variables: ["tdoa3.snr_1", "tdoa3.powerdiff_1", "tdoa3.snr_2", "tdoa3.powerdiff_2"]
, the frequency of  "/cf1/log1"  is around 20 Hz and the frequency of "/cf1/log2" is around 50 Hz.

Also, I tried to tune the "genericLogTopicFrequencies" values, yet the logging frequencies are not the values set there. 

The "genericLogTopicFrequencies" values are sent to the "logBlock.frequency" in  "crazyswarm_server.cpp" which seems to control the logging blocks. However, the logging frequencies do not match the setting values 

I am wondering 
(1) What is the unit of "genericLogTopicFrequencies" in "hover_swarm.launch"? Does it control the logging frequency?
(2) If I wish to log two topics with each topic 50 Hz, how should I set the parameter in launch file?

Any suggestions are appreciated. 
Thanks.

**Comment by whoenig** [comment, marked as answer]
1. It's very poor naming. It's actually the period in ms, so 10 means 100 Hz.
2. Technically you would set both to 20. If you still don't get enough data, you might need to reduce the waiting time in https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/src/crazyswarm_server.cpp#L1016. (The Crazyswarm is not optimized for logging and logging is usually only as tool for debugging of a single CF.)

**Reply by Williamwenda** [reply]
Thanks for the quick reply. Now I understand the ROS frequencies I saw. 
Fortunately, I am using the logging only for debugging a single CF. For a large amount of data-logging, I will use the SD-card. 

Very appreciated your explanation.
